packages/enhanced/src/lib/sharing/ConsumeSharedPlugin.ts

```diff
@@ -259,15 +258,17 @@ class ConsumeSharedPlugin {
                     data,
                     packageName,
                   );
-                  if (typeof requiredVersion !== 'string') {
-                    requiredVersionWarning(
-                      `Unable to find required version for "${packageName}" in description file (${descriptionPath}). It need to be in dependencies, devDependencies or peerDependencies.`,
-                    );
-                    return resolve(undefined);
-                  }
                   // @ts-ignore  webpack internal semver has some issue, use runtime semver , related issue: https://github.com/webpack/webpack/issues/17756
                   resolve(requiredVersion);
                 },
+                ({ data }) => {
+                  const maybeRequiredVersion =
+                    getRequiredVersionFromDescriptionFile(data, packageName);
+                  return (
+                    data['name'] === packageName ||
```

Comment:
if no version warning should still get logged

Reply:
same warning will be registered

Reply:
but i see this messge removed 

 `Unable to find required version for "${packageName}" in description file (${descriptionPath}). It need to be in dependencies, devDependencies or peerDependencies.`,
                    );
                    
                    will same message report or does it fall back to unable to find description file only?

Reply:
Oh yeah, I thought about the self-referencing warning that doesn't get registered.  
https://github.com/module-federation/core/blob/5ac093d812defef3672695a22bf5455962fde4ea/packages/enhanced/src/lib/sharing/ConsumeSharedPlugin.ts#L256  
  
Maybe accumulate warnings for all description file names (`descriptionPath `) visited during recursive lookup to create a summary warning?

Reply:
Good idea, if you want to update the PR to do so, feel free. Otherwise just make sure that it reports back that warning if it comes up dry, rather than removing it completely. Its useful to tell dev they should list the package in their deps etc if it cant find one, especially now that it can recurse upwards

Reply:
All good. I'll pull locally and touch it up a little then merge

Reply:
It's okay, it's my mistake - I didn't explain it well☺️